api/handlers/dispute.go

```diff
@@ -223,37 +231,69 @@ func (h Dispute) getDispute(c *gin.Context) {
 		Role:        role,
 	}
 
-	err = h.DB.Raw("SELECT file_name,uploaded,file_path FROM files WHERE id IN (SELECT file_id FROM dispute_evidence WHERE dispute = ?)", id).Scan(&DisputeDetailsResponse.Evidence).Error
+	evidenceQuery := `
+		SELECT 
+			f.id,
+			f.file_name,
+			f.uploaded,
+			f.file_path,
+			CASE
+				WHEN d.complainant = de.user_id THEN 'Complainant'
+				WHEN d.respondant = de.user_id THEN 'Respondent'
+				ELSE 'Other'
+			END AS uploader_role
+		FROM 
+			dispute_evidence de
+		JOIN 
+			files f ON de.file_id = f.id
+		JOIN 
+			disputes d ON de.dispute = d.id
+		WHERE 
+			de.dispute = ?
+	`
+
+	var evidence []models.Evidence
+	err = h.DB.Raw(evidenceQuery, id).Scan(&evidence).Error
```

Comment:
why you rawdogging the query? What's wrong with using functions like `Select`, `Joins` and `WHERE`?